File: Data Structures & Algorithms/validate-parentheses/submission-2.rs
impl Solution {
    pub fn is_valid(s: String) -> bool {
        let mut stack: Vec<char> = vec![];

        for b in s.chars() {
            if b == '(' || b == '{' || b == '[' {
                stack.push(b);
                continue
            };
            
            if let Some(last) = stack.pop() {
                if last == '(' && b == ')' || last == '{' && b == '}' || last == '[' && b == ']' {
                    continue
                }
            }

            return false
        }

        return stack.len() == 0
    }
}

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-2.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        l, r = 0, 0
        
        while r < len(s):
            if s[r + 1] == s[l]:
                l += 1
                r += 1
            else:
                r += 1
        
        return r - l
